frontend/src/components/common/PrintCSVButton.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import {
+  Icon,
+  IconButton,
+  Box,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  FormControl,
+  Button,
+  FormHelperText,
+  Flex,
+  ScaleFade,
+  Alert,
+  AlertDescription,
+  AlertIcon,
+} from "@chakra-ui/react";
+import { AiFillPrinter } from "react-icons/ai";
+import { RangeDatepicker } from "chakra-dayzed-datepicker";
+import commonAPIClient from "../../APIClients/CommonAPIClient";
+import CSVConverter from "../../helper/CSVConverter";
+
+const PrintCSVButton = (): React.ReactElement => {
+  const [selectedDates, setSelectedDates] = useState<Date[]>([]);
+  const [showAlert, setShowAlert] = useState(false);
+
+  const handleClear = () => {
+    setSelectedDates([]);
+  };
+
+  const [isOpen, setOpen] = useState(false);
+
+  const handleOpen = () => {
+    setOpen(true);
+    setSelectedDates([]);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleSubmit = async () => {
+    // Use YYYY-MM-DD format
+    const dateRange = selectedDates.map((date) =>
+      date.toISOString().substring(0, 10),
+    );
+
+    const data = await commonAPIClient.filterLogRecords(
+      undefined,
+      undefined,
+      undefined,
+      dateRange,
+      undefined,
+      undefined,
+      true, // return all data
+    );
+
+    setShowAlert(CSVConverter(data.logRecords) === "error");
+  };
+
+  useEffect(() => {
+    if (showAlert) {
+      setTimeout(() => {
+        setShowAlert(false);
+      }, 3000);
+    }
+  }, [showAlert]);
+
+  return (
+    <>
+      <IconButton
+        aria-label="Print CSV"
+        className="ghost-button"
+        icon={<Icon boxSize="32px" as={AiFillPrinter} />}
+        variant="ghost"
+        onClick={handleOpen}
+      />
+
+      <Box>
+        <Modal isOpen={isOpen} onClose={handleClose} size="xl">
+          <ModalOverlay />
+          <ModalContent>
+            <ModalHeader>Print CSV File</ModalHeader>
+            <ModalBody>
+              <FormControl>
+                <Flex gap="8px">
+                  <RangeDatepicker
+                    selectedDates={selectedDates}
+                    onDateChange={setSelectedDates}
+                  />
```

Comment:
One thing I think that might be more beneficial in our case is simply copy-pasting this from https://github.com/aboveyunhai/chakra-dayzed-datepicker/tree/main/src/components and leveraging that instead for increased customizability going forward (owner of the pkg also recommends this)

Reply:
I think its okay to leave it the way it is for now, there is a new ticket to style the date picker (#86) so I can include the link above in that ticket